holoviews/plotting/mpl/chart.py

```diff
@@ -1258,3 +1258,78 @@ class SideSpikesPlot(AdjoinedPlot, SpikesPlot):
         all axis labels including ticks and ylabel. Valid options are 'left',
         'right', 'bare' 'left-bare' and 'right-bare'.""")
 
+
+
+class SegmentPlot(ColorbarPlot):
+    """
+    Segments are lines in 2D space where each two each dimensions specify a
+    (x, y) node of the line.
+    """
+    # Deprecated parameters
+
+    color_index = param.ClassSelector(default=None, class_=(basestring, int),
+                                      allow_None=True, doc="""
+        Deprecated in favor of color style mapping, e.g. `color=dim('color')`""")
+
+    size_index = param.ClassSelector(default=None, class_=(basestring, int),
+                                     allow_None=True, doc="""
+        Deprecated in favor of size style mapping, e.g. `size=dim('size')`""")
```

Comment:
Let's not add these deprecated parameters to new elements.

Reply:
Agreed, see above.